File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Entities/Training.java
package com.example.tfg_urjc_appfirstrun.Entities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.ArrayList;
import java.util.Date;
import java.util.UUID;

@Entity (tableName = "trainings")
public class Training {
    @PrimaryKey
    @NonNull
    public String trainingId;
    @ColumnInfo(name = "name")
    public String name;
    @ColumnInfo(name = "start_date")
    public Date startDate;
    @ColumnInfo(name = "is_Finished")
    public boolean isFinished;
    @ColumnInfo(name = "type_training")
    public String typeTraining;
    @ColumnInfo(name = "mark_5km")
    public String mark5Km;

    public Training () {
        this.trainingId = UUID.randomUUID().toString();
        this.name = null;
        this.startDate = null;
        this.isFinished = false;
        this.typeTraining = null;
        this.mark5Km = null;
    }

    public Training(String name, Date startDate, String typeTraining, String mark5Km){
        this.trainingId = UUID.randomUUID().toString();
        this.name = name;
        this.startDate = startDate;
        this.isFinished = false;
        this.typeTraining = typeTraining;
        this.mark5Km = mark5Km;
    }

    public String getTrainingId() {
        return trainingId;
    }
}

File: app/src/main/java/com/example/tfg_urjc_appfirstrun/Entities/Training.kt
package com.example.tfg_urjc_appfirstrun.Entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.jetbrains.annotations.NotNull
import java.util.*

@Entity(tableName = "trainings")
class Training(name: String?, startDate: Date?, typeTraining: String?, mark5Km: String?){
    @PrimaryKey
    @NotNull
    var trainingId: String

    @ColumnInfo(name = "name")
    var name: String?

    @ColumnInfo(name = "start_date")
    var startDate: Date?

    @ColumnInfo(name = "is_Finished")
    var isFinished: Boolean

    @ColumnInfo(name = "type_training")
    var typeTraining: String?

    @ColumnInfo(name = "mark_5km")
    var mark5Km: String?

    init {
        trainingId = UUID.randomUUID().toString()
        this.name = name
        this.startDate = startDate
        isFinished = false
        this.typeTraining = typeTraining
        this.mark5Km = mark5Km
    }
}